test/custom_task_test.go

```diff
@@ -51,10 +51,18 @@ const (
 	kind       = "Wait"
 )
 
-var supportedFeatureGates = map[string]string{
-	"enable-custom-tasks": "true",
-	"enable-api-fields":   "alpha",
-}
+var (
+	supportedFeatureGates = map[string]string{
+		"enable-custom-tasks": "true",
+		"enable-api-fields":   "alpha",
+	}
+
+	filterTypeMeta          = cmpopts.IgnoreFields(metav1.TypeMeta{}, "Kind", "APIVersion")
```

Comment:
Why do we filter `Kind` and `APIVersion`? 

Makes sense to filter the random fields, such as `CreationTimestamp` and `CompletionTime`, but it's not clear why we filter these fields that are known

Reply:
The actual resources retrieved from API server don't specify Kind and APIVersion. Do you think It's better to manually set the two fields in wantResource (like wantPipelineRun or wantRun) as empty strings?

Reply:
ah makes sense, checked that other tests also do the same, thanks!